src/nanopolish_extract.cpp

```diff
@@ -145,8 +145,16 @@ void process_file(const std::string& fn)
             auto fq = f.get_basecall_fastq(p.first, p.second);
             auto fq_a = f.split_fq(fq);
             // construct name
-            auto pos = fq_a[0].find_first_of('_');
-            std::string name = fq_a[0].substr(0, pos) + ":" + p.second + ":";
+            std::string name;
+            std::istringstream(fq_a[0]) >> name;
+            std::string::size_type pos = 0;
+            while (true)
```

Comment:
`std::replace(name.begin(), name.end(), ':', '_')` from <algorithms> would be simpler I think